typings/glamorous.d.ts

```diff
@@ -40,6 +40,12 @@ interface ThemeProps {
 
 export class ThemeProvider extends React.Component<ThemeProps, any> { }
 
+export declare function withTheme
+  <P>(
+    component:
+      | React.ComponentClass<P>
```

Comment:
This would fail when user tries to use `theme` prop. To avoid that, it could be written as

```
export type Component<T> = React.ComponentClass<T> | React.StatelessComponent<T>;

// `ThemeProps` comes from https://github.com/paypal/glamorous/blob/master/typings/glamorous.d.ts#L37-L39.
export declare function withTheme<P>(component: Component<P & ThemeProps>): GlamorousComponent<P>;

```